README.md

```diff
@@ -22,16 +23,113 @@ Then you can build it just with:
 
 ```
 cd holochain-rust
-cargo build
+cargo build --release
 ```
 
 given that you have rust installed.
 
-Now either open and build holosqape.pro with [Qt Creator](https://en.wikipedia.org/wiki/Qt_Creator) (esp. if you want to build for Android or iOS) or run:
+Now either open and build all.pro with [Qt Creator](https://en.wikipedia.org/wiki/Qt_Creator) (esp. if you want to build for Android or iOS) or run:
 
 ```
 qmake
 make
 ```
 
-![](images/screenshots/mainwindow.png)
+Make sure you run matching build configurations (i.e. debug/release). The commands above are for release. If you want to build debug do:
+
+```
+cd holochain-rust
+cargo build
+cd ..
+qmake CONFIG+=debug
+make
+```
+
+## Sub directories
+This repository consist of several sub directories containing separate sub projects.
+
+* **holochain-rust** contains the Holochain repository as a sub-module. This project compiles to a static library.
+* **bindings** contains Qt/C++ wrappers around the Holochain Rust code that is made available in the form of plain C functions and structs. This project makes Holochain available as Qt-based QObjects that can easily be plugged into Qt script engines. This project compiles to a static library.
+* **holosqape** contains the QML based GUI project wich compiles to an executable that can be build for all platforms supported by Qt. It manages installed Holochain apps and currently sports a generic UI that creates widgets and buttons for each zome function an app implements automatically. It also includes a JavaScript based console in which apps can be instantiated.
+* **holoconsole** this builds to another executable that basically hosts the same functionality as holosqape but without GUI. Instead it has some compatibility shims to enable running JavaScript libraries such as [Tape](https://github.com/substack/tape). It's main purpose is hosting an environment for app tests, including scenario integration tests.
+* **app-test-scaffold** this is an example showing how to build an app test script to run with holoconsole.
+
+
+
+## App tests
+
+### Binding objects
+In order to test a Holochain app we need a JS script that we run in holoconsole. The script engine adds a global object ```Container``` to the context before loading and running the user script. It currently has the following methods:
+
+* installedApps()
+* instantiate(dnaHash)
+* installApp(filename)
+
+and more as can be seen in [container.h](bindings/container.h).
+
+instantiate() returns an ```App``` object as defined in [app.h](bindings/app.h) with a method
```

Comment:
Please use single backticks for inline code and triple backticks for code blocks